### PR TITLE
Loosened up type checking for Tabs components

### LINKED ISSUE #539: TabList and TabPanels do not comply with using lists.map to loop over Tab and TabPanel
**Describe the bug**

The Tabs component with TabList or TabPanels in combination with a **<objectlist>.map** to render a set of _Tab_ or a set of _TabPanel_ creates a forwardRef warning in devTools.
_Warning: Failed prop type: Invalid prop `children` supplied to `ForwardRef(TabLists)`._
_Warning: Failed prop type: Invalid prop `children` supplied to `ForwardRef(TabPanels)`._

To use a map over a list of objects is convenient when you have a long list, or the length is dynamic.

**Steps to reproduce the bug**
code example
```
const items = [ {object A}, {object B}, {object C}]

<Tabs>
  <TabList>
     {items.map(item => (
         <Tab>{item.name}</Tab>
     )}
  </TabList>
  <TabPanels>
    {items.map(item=>(
      <TabPanel>{item.value}</TabPanel>
    )}
  </TabPanels>
</Tabs>

```

**Expected behavior**
the same as if not using a map

```
const items = [ {object A}, {object B}, {object C}]

<Tabs>
  <TabList>
     <Tab>{items[0].name}</Tab>
     <Tab>{items[1].name}</Tab>
     <Tab>{items[2].name}</Tab>
  </TabList>
  <TabPanels>
     <TabPanel>{items[0].value}</TabPanel>
     <TabPanel>{items[1].value}</TabPanel>
     <TabPanel>{items[2].value}</TabPanel>
  </TabPanels>
</Tabs>

```

### PR DESCRIPTION
**Pull request by mimarz**
fixes #539

When using Styled Components, Emotion or any other library who wraps the React component for styling changes type when checking on propTypes.

Loosened up checking so that its possible to use these to style compound components in `Tabs`

### CODE REVIEW

File: apps/storybook-react/stories/Tabs.stories.jsx [hunk 1]
```diff
@@ -201,3 +201,45 @@ export const tabsAndInputInPanel = () => {
     </div>
   )
 }
+
+const StyledTab = styled(Tab)`
+  background: pink;
```
Comment (vnys) [inline apps/storybook-react/stories/Tabs.stories.jsx:206]: 💅

File: apps/storybook-react/stories/Tabs.stories.jsx [hunk 2]
```diff
@@ -201,3 +201,45 @@ export const tabsAndInputInPanel = () => {
...
+`
+
+const StyledTabPanel = styled(TabPanel)`
+  padding: 32px;
+  background: peachpuff;
```
Comment (vnys) [inline apps/storybook-react/stories/Tabs.stories.jsx:211]: 💋

File: libraries/core-react/src/Tabs/TabPanels.jsx [hunk 3]
```diff
@@ -1,35 +1,33 @@
...
 import { TabsContext } from './Tabs.context'
-import { TabPanel } from './TabPanel'
 
 const TabPanels = forwardRef(function TabPanels({ children, ...props }, ref) {
   const { activeTab, tabsId } = useContext(TabsContext)
 
-  const Panels = React.Children.map(children, (child, index) =>
-    React.cloneElement(child, {
+  const Panels = React.Children.map(children, (child, index) => {
+    console.log(child.type)
```
Comment (vnys) [inline libraries/core-react/src/Tabs/TabPanels.jsx:9]: 🙀

**Review by vnys — APPROVED** [review]
One`console.log` we should remove, but apart from that it LGTM 👍

File: libraries/core-react/src/Tabs/TabPanels.jsx (hunk 3 above)
Comment (mimarz) [reply libraries/core-react/src/Tabs/TabPanels.jsx:9]: 🙊

**Pull request merged**